fsm/fsm.go

```diff
@@ -69,35 +40,25 @@ type CmdStateTuple struct {
 }
 
 // TransitionFunc models a transition function
-type TransitionFunc func(m *FSM) interface{}
+type TransitionFunc func(m *FSM) (string, []Message)
```

Comment:
Due to the change of the fsm `Function` we no longer have to pass around an interface as a value.